### PR TITLE
PN-4642 enhance dates control in notification filters

### PR DESCRIPTION
**Pull request by clombardi**
## Short description
Make formik validations (which control the "filtra" button enable status) coherent with input field restrictions (which control the color of the frame around fields).  
Besides this, force both startDate and endDate to be >= tenYearsAgo.

## List of changes proposed in this pull request
- Define custom tests for startDate and endDate in the Formik specification.
- Add the ">= tenYearsAgo" restriction to the startDate and endDate input fields in the form body definition.
- Add some tests regarding the Formik validations.

## How to test
Play with notification list filters, in all of PA / PF / PG.

### CODE REVIEW

File: packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx [hunk 1]
```diff
@@ -69,6 +69,33 @@ function isFilterapplied(filtersCount: number): boolean {
 
 const getValidStatus = (status: string) => (status === 'All' ? '' : status);
 
+// the formik validations (which control the enable status of the "filtra" button)
+// must coincide with the input field validations (which control the color of the frame around each field)
+const dateFieldsValidation = {
+  // startDate cannot be earlier than 10 years ago
+  startDate: yup.date().min(tenYearsAgo).test({
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx:76]: why not yup.date().min(tenYearsAgo).max(today) ?

File: packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx [hunk 2]
```diff
@@ -69,6 +69,33 @@ function isFilterapplied(filtersCount: number): boolean {
...
+    name: 'notInFuture-start',
+    test(value) {
+      return !value || value.getTime() <= today.getTime();
+    }
+  }),
+  // endDate cannot be earlier than 10 years ago
+  endDate: yup.date().min(tenYearsAgo).test({
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx:83]: The same of startDate.

File: packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx [hunk 3]
```diff
@@ -69,6 +69,33 @@ function isFilterapplied(filtersCount: number): boolean {
...
+    name: 'notInFuture-end',
+    test(value) {
+      return !value || value.getTime() <= today.getTime();
+    }
+  // endDate cannot be earlier than start date
+  }).test({
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx:89]: We can use the min condition. In the component we have two state variables startDate and endDate.
So we can do something like that 

```
endDate: yup
      .date()
      .min(startDate && !isNaN(startDate.getTime()) ? startDate : tenYearsAgo)
```
This code increase the complexity, so we can define a function

`
const dateIsDefined = (date: Date | null) => date && !isNaN(date.getTime());
`

in date.utility.ts

File: packages/pn-personafisica-webapp/src/component/Notifications/FilterNotifications.tsx [hunk 4]
```diff
@@ -71,6 +71,34 @@ const initialValues = (
   };
 };
 
+
+// the formik validations (which control the enable status of the "filtra" button)
+// must coincide with the input field validations (which control the color of the frame around each field)
+const dateFieldsValidation = {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Notifications/FilterNotifications.tsx:77]: Same of before

File: packages/pn-personagiuridica-webapp/src/component/Notifications/FilterNotifications.tsx [hunk 5]
```diff
@@ -71,6 +71,34 @@ const initialValues = (
   };
 };
 
+
+// the formik validations (which control the enable status of the "filtra" button)
+// must coincide with the input field validations (which control the color of the frame around each field)
+const dateFieldsValidation = {
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/component/Notifications/FilterNotifications.tsx:77]: Same as before

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx (hunk 1 above)
Comment (clombardi) [reply packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx:76]: ok, done as suggested

File: packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx (hunk 2 above)
Comment (clombardi) [reply packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx:83]: ok, done as suggested

File: packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx (hunk 3 above)
Comment (clombardi) [reply packages/pn-pa-webapp/src/pages/components/Notifications/FilterNotifications.tsx]: ok done as suggested

File: packages/pn-personafisica-webapp/src/component/Notifications/FilterNotifications.tsx (hunk 4 above)
Comment (clombardi) [reply packages/pn-personafisica-webapp/src/component/Notifications/FilterNotifications.tsx]: ok done as before

File: packages/pn-personagiuridica-webapp/src/component/Notifications/FilterNotifications.tsx (hunk 5 above)
Comment (clombardi) [reply packages/pn-personagiuridica-webapp/src/component/Notifications/FilterNotifications.tsx]: ok done as before

**Pull request merged**